readme.md

```diff
@@ -1,48 +1,230 @@
+
+## Contents
+- [Installation](#installation)
+- [Usage](#usage)
+  - [The scroll parent](#the-scroll-parent)
+  - [Scroll triggers](#scroll-triggers)
+  - [Examples](#examples)
+    - [Parallax scrolling](#parallax-scrolling)
+    - [Playing video/animation on scroll](#playing-videoanimation-on-scroll)
+    - [Scroll reveals](#scroll-reveals)
+    - [Custom triggers](#custom-triggers)
+  - [Usage with React](#usage-with-react)
+  - [Usage with `react-spring`](#usage-with-react-spring)
+- [Examples](#examples)
+- [Tips & Gotcha's](#tips--gotchas)
+  - [Do NOT use `vh` units in your page](#do-not-use-vh-units-in-your-page)
+  - [Optimize performance with css `contain`](#optimize-performance-with-css-contain)
+  - [Transforms](#transforms)
+  - [Track horizontal scrolling](#track-horizontal-scrolling)
 
 ## Installation
 
 ```
-yarn add @kaliber/library
+yarn add @kaliber/scroll-progression
 ```
 
+#### Transpilation
+
+When working with `@kaliber/build`, add `/@kaliber\/scroll-progression/` to your `compileWithBabel` array. 
+
+#### Polyfills
+
+- `ResizeObserver`
+
+---
+
 ## Usage
-Short example. If your library has multiple ways to use it, show the most used one and refer to `/example` for further examples.
 
-```jsx
-import { hello } from 'library'
+This library tracks to progression of a given element between two points within its [scroll parent](#the-scroll-parent). These two points are called [scroll triggers](#scroll-triggers). 
+
+### The scroll parent
+The scroll parent of an element is found by finding the closest parent that has an `overflow` or `overflow-y` value of `auto` or `scroll`.
+
+### Scroll triggers
+
+All the functions defined in `@kaliber/scroll-progression/triggers` return an object consisting of a `fraction` and (optionally) an `offset`. The `fraction` is a relative value based on the scroll parent's height. `0` means the top of the scroll parent, `1` means the bottom. If you need to increase or decrease with an amount of pixels, you can use the `offset` property for this. 
+
+For instance: a scroll trigger describing the point 100 pixels below the top of an element is described as follows:
+
+```js
+{ fraction: 0, offset: 100 }
+```
+
+To improve readability, a set of triggers is exported which describe the most common scroll triggers:
+
+| Function  |   |
+|---|---|
+| top  | `(offset = 0) => ({ fraction: 0, offset })`  |
+| center  | `(offset = 0) => ({ fraction: 0.5, offset: 0 })`  |
+| bottom  | `(offset = 0) => ({ fraction: 1, offset: 0 })`  |
+| custom  | `(fraction, offset = 0) => ({ fraction, offset })`  |
+
+Below are some examples to help you get a feeling for this. To view some examples in a practical setting, check the [`/example`](https://github.dev/kaliberjs/scroll-progression) folder!
+
+### Examples
```

Comment:
Ik zou hier nog een klein stukje tussen zetten waarbij je uitlegt wat de signature van de functie doet, zoiets:

```js
onScrollProgression({
   // The element we are determining the scroll for
  element,

  // The starting position (everything before this point results in 0)
  start: {
    // The start location relative to the element (in this case the top of the element)
    element: { fraction: 0, offset: 0 }, 

    // The start location relative to the scroll parent (in this case the top of the scroll parent)
    scrollParent: { fraction: 0, offset: 0 },
  },

  // The end position (everything after this point results in 1)
  end: {
    // The end location relative to the element (in this case the bottom of the element)
    element: { fraction: 1, offset: 0 },

    // The end location relative to the element (in this case the bottom of the scroll parent)
    scrollParent: { fraction: 1, offset: 0 },
  },

  // Callback, called when the progression (a value between 0 and 1) changes
  onChange(progression) {
    ...
  }
})
```

In the above example the progression will go from, hmmm, ik snap het niet, haha.

Reply:
@peeke Je moet maar even kijken of het ook echt klopt.

Reply:
Het klopt helemaal :+1: